examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,397 @@
+#
+# Instead of relying solely on `max_iter` to determine when to stop, the HGBT
+# implementations in scikit-learn support early stopping. With it, the model
+# uses a fraction of the training data as a validation set
+# (`validation_fraction`) and stops training if the validation score does not
+# improve (or degrades) after `n_iter_no_change` iterations up to a certain
+# `tol`.
+#
+# Notice that there is a trade-off between `learning_rate` and `max_iter`:
+# Generally, smaller learning rates require more iterations to converge to the
+# minimum loss, while larger learning rates might converge faster but are at
+# risk of overfitting.
+#
+# Indeed, a good practice is to tune the learning rate along with any other
+# hyperparameters, fit the HBGT on the training set with a large enough value
+# for `max_iter` and determine the best `max_iter` via early stopping and some
+# explicit `validation_fraction`.
+
+common_params = {
+    "max_iter": 1_000,
+    "learning_rate": 0.3,
+    "validation_fraction": 0.2,
+    "random_state": 42,
+    "scoring": "neg_root_mean_squared_error",
+}
+
+hgbt = HistGradientBoostingRegressor(early_stopping=True, **common_params)
+hgbt.fit(X, y)
+plt.plot(-hgbt.validation_score_)
+plt.xlabel("number of iterations")
+plt.ylabel("root mean squared error")
+_ = plt.title(f"Loss of hgbt with early stopping (n_iter={hgbt.n_iter_})")
+
+# %%
+# We can then overwrite the value for `max_iter` to a razonable value and avoid
+# the extra computational cost of the inner validation. In this case, rounding
+# up the number of iterations to 600 may account for variability of the training
+# set:
+
+common_params["max_iter"] = 600
+common_params["early_stopping"] = False
+hgbt = HistGradientBoostingRegressor(**common_params)
+
+# %%
+# .. note:: The inner validation done during early stopping is not optimal for
+#    time series with the implementation as of scikit-learn v1.3.
+#
+# Support for missing values
+# ==========================
+# HGBT models have native support of missing values. During training, the tree
+# grower decides where samples with missing values should go (left or right
+# child) at each split, based on the potential gain. When predicting, these
+# samples are sent to either child accordingly. If a feature had no missing
+# values during training, samples with missing values for that feature are sent
+# to the child with the most samples.
+#
+# The present example shows how HGBT regressions deal with values missing
+# completely at random (MCAR), i.e. the missingness does not depend on the
+# observed data or the unobserved data. We can simulate such scenario by
+# randomly replacing values from randomly selected features with `Nan` values.
+
+import numpy as np
+
+from sklearn.model_selection import TimeSeriesSplit, cross_validate
+
+np.random.seed(42)
+
+ts_cv = TimeSeriesSplit(n_splits=5, gap=48, max_train_size=10000, test_size=1000)
+train_0, test_0 = next(ts_cv.split(df))
+last_days = slice(-192, None)
```

Comment:
Instead to have a test set and another last days test set. I am wondering if we can just have the train set and the last days set since we are only using this for the plotting.
I think it will simplify the code below.